HorizonEngine/Platform/OpenGL/GLFrameBuffer.cpp

```diff
@@ -0,0 +1,76 @@
+#include "pch.h"
+#include "GLFrameBuffer.h"
+
+#include "glad/glad.h"
+#include "GLFW/glfw3.h"
+
+namespace Hzn
+{
+	GLFrameBuffer::GLFrameBuffer(const FrameBufferProps& props)
+		: m_Props(props)
+	{
+		invalidate();
+	}
+
+	GLFrameBuffer::~GLFrameBuffer()
+	{
+		destroy();
+	}
+
+	void GLFrameBuffer::destroy()
+	{
+		glDeleteFramebuffers(1, &m_FrameBufferId);
+		glDeleteTextures(1, &m_ColorAttachment);
+		glDeleteTextures(1, &m_DepthAttachment);
+	}
+
+
+	void GLFrameBuffer::invalidate()
+	{
+		if(m_FrameBufferId != 0)
+		{
+			destroy();
+		}
+
+		glGenFramebuffers(1, &m_FrameBufferId);
+		glBindFramebuffer(GL_FRAMEBUFFER, m_FrameBufferId);
+
+		// Color attachment
+		glGenTextures(1, &m_ColorAttachment);
```

Comment:
If the GLFrameBuffer is being invalidated why are we attaching colours, textures etc. with in this invalidation

Reply:
Because we want to resubmit that data, when we are recreating the framebuffer. Frame buffers need to have color and depth attachments for rendering.